modules/mongodb/src/main/java/org/testcontainers/containers/MongoDbContainer.java

```diff
@@ -0,0 +1,137 @@
+
+/**
+ * Constructs a single node MongoDB replica set for testing transactions.
+ * <p>To construct a multi-node MongoDB cluster, consider the <a href="https://github.com/silaev/mongodb-replica-set/">mongodb-replica-set project on GitHub</a>
+ * <p>Tested on a Mongo DB version 4.0.10+ (that is the default version if not specified).
+ *
+ * @author Konstantin Silaev on 9/30/2019
+ */
+@Slf4j
+public class MongoDbContainer extends GenericContainer<MongoDbContainer> {
+    private static final int CONTAINER_EXIT_CODE_OK = 0;
+    static final int MONGODB_INTERNAL_PORT = 27017;
+    private static final int AWAIT_INIT_REPLICA_SET_ATTEMPTS = 30;
+    private static final String MONGODB_VERSION_DEFAULT = "4.0.10";
+    private static final String MONGODB_DATABASE_NAME_DEFAULT = "test";
+
+    public MongoDbContainer() {
+        super("mongo:" + MONGODB_VERSION_DEFAULT);
+        configureMongoDbContainer();
+
+    }
+
+    public MongoDbContainer(@NonNull final String dockerImageName) {
+        super(dockerImageName);
+        configureMongoDbContainer();
+    }
+
+    private void configureMongoDbContainer() {
+        withExposedPorts(MONGODB_INTERNAL_PORT);
+        withCommand("--replSet", "docker-rs");
+        waitingFor(
+            Wait.forLogMessage(".*waiting for connections on port.*", 1)
+        );
+    }
+
+    public String getReplicaSetUrl() {
+        if (!isRunning()) {
+            throw new IllegalStateException("MongoDbContainer should be started first");
+        }
+        return String.format(
+            "mongodb://%s:%d/%s",
+            getContainerIpAddress(),
+            getMappedPort(MONGODB_INTERNAL_PORT),
+            MONGODB_DATABASE_NAME_DEFAULT
+        );
+    }
+
+    @Override
+    public void start() {
+        super.start();
+        initReplicaSet();
+        logReplicaSetStatus();
+    }
+
+    @SneakyThrows(value = {IOException.class, InterruptedException.class})
+    private void logReplicaSetStatus() {
+        log.debug(
+            "REPLICA SET STATUS:\n{}",
+            execInContainer(buildMongoEvalCommand("rs.status()")).getStdout()
+        );
+    }
+
+    private String[] buildMongoEvalCommand(final String command) {
+        return new String[]{"mongo", "--eval", command};
+    }
+
+    private void checkMongoNodeExitCode(final Container.ExecResult execResult) {
+        if (execResult.getExitCode() != CONTAINER_EXIT_CODE_OK) {
+            final String errorMessage = String.format("An error occurred: %s", execResult.getStdout());
+            log.error(errorMessage);
+            throw new ReplicaSetInitializationException(errorMessage);
+        }
+    }
+
+    private String buildMongoWaitCommand() {
+        return String.format(
+            "var attempt = 0; " +
+                "while" +
+                "(%s) " +
+                "{ " +
+                "if (attempt > %d) {quit(1);} " +
+                "print('%s ' + attempt); sleep(1000);  attempt++; " +
```

Comment:
1 second delay is too big, please change it to something lower (100ms maybe)

Reply:
done

Reply:
LGTM